app/bundles/CampaignBundle/Controller/CampaignMapStatsController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Mautic\CampaignBundle\Controller;
+
+use Doctrine\DBAL\Exception;
+use Mautic\CampaignBundle\Entity\Campaign;
+use Mautic\CampaignBundle\Model\CampaignModel;
+use Mautic\CoreBundle\Helper\MapHelper;
+use Mautic\CoreBundle\Security\Permissions\CorePermissions;
+use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+class CampaignMapStatsController extends AbstractController
+{
+    public const MAP_OPTIONS = [
+        'contacts' => [
+            'label' => 'mautic.lead.leads',
+            'unit'  => 'Contact',
+        ],
+        'read_count' => [
+            'label' => 'mautic.email.read',
+            'unit'  => 'Read',
+        ],
+        'clicked_through_count'=> [
+            'label' => 'mautic.email.click',
+            'unit'  => 'Click',
+        ],
+    ];
+
+    public const LEGEND_TEXT = 'Total: %total (%withCountry with country)';
+
+    public function __construct(protected CampaignModel $model)
```

Comment:
Any reason for `protected`?
```suggestion
    public function __construct(private CampaignModel $model)
```